### DISCUSSION TITLE
Need help: usage error in integrating vault with vector

### DISCUSSION THREAD

**Discussion by puppet-py** [vectordotdev/vector#17539] [Q&A] [answered]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

Following the instruction as stated in these:
 https://vector.dev/docs/reference/configuration/global-options/#secret
https://github.com/vectordotdev/vector/blob/aa45cefabc165cce5cc2135ad8b23705b524b769/website/content/en/highlights/2022-07-07-secrets-management.md

The intent is to give Vector access: `vault > specific namespace > folder > <secrets>
`
Not sure where and how the `secret.exec` or `secret.exec.command` to be used. 

For example: the secrets are stored in the VAULT= `https://mystg.vault.net`, inside namespace `dev`, folder `mysecrets` , `mykey`

looks like we are doing an incorrect usage in order to fetch the secrets from Vault. Any guidance and correction are much appreciated on how to set up the configuration file such that Vector can access the Vault.

### Version

0.29.1

**Comment by puppet-py** [comment]
@jszwedko seeking your attention :-)

**Reply by jszwedko** [reply, marked as answer]
(converted this to a discussion since it didn't seem like an issue report)

Hi @puppet-py !

Vector doesn't have direct integration with Vault. What it does have is the ability to execute a command to fetch secrets. How those secrets are fetched are up to you. I imagine you could write a bash script that uses the Vault CLI to to fetch secrets and return them in the format that Vector expects them (as documented in https://github.com/vectordotdev/vector/blob/aa45cefabc165cce5cc2135ad8b23705b524b769/website/content/en/highlights/2022-07-07-secrets-management.md.

If you'd like a direct Vault integration, could you open a feature request?
